src/components/SaveEventButton.js

```diff
@@ -57,16 +49,29 @@ export default class SaveEventButton extends React.Component<Props, State> {
     );
   }
 
-  componentDidUpdate(prevProps: Props) {
-    // Only animate if props.active has changed,
-    // Animates heart when change from inactive -> active
-    // Snaps to start when change from active -> inactive
-    if (this.state.progress && this.props.active !== prevProps.active) {
-      triggerAnimation(this.state.progress, this.props.active);
+  startAnimation = (value: number) => {
+    if (this.state.progress) {
+      Animated.timing(this.state.progress, {
+        toValue: value,
+        duration: value * 800,
+        easing: Easing.linear,
+        useNativeDriver: true
+      }).start(this.completeAnimation);
+
+      if (value) {
+        ReactNativeHapticFeedback.trigger("impactHeavy");
+      }
+
+      this.setState({ animating: true });
     }
-  }
+  };
+
+  completeAnimation = () => {
+    this.setState({ animating: false });
```

Comment:
That could crash when the component was unmounted while the animation is running.

Reply:
Yeah, I added a test for this and fixed it.